pkg/image/imageutil/helpers.go

```diff
@@ -286,10 +296,138 @@ func ImageWithMetadataOrDie(image *imagev1.Image) {
 
+// TagReferencesLocalTag returns true if the provided tag reference references another image stream tag
+// in the current image stream. This is only true when from points to an ImageStreamTag without a colon
+// or from.name is <streamName>:<tag>.
+func TagReferencesLocalTag(stream *imagev1.ImageStream, tag imagev1.TagReference) (string, bool) {
+	if tag.From == nil || tag.From.Kind != "ImageStreamTag" {
+		return "", false
+	}
+	if len(tag.From.Namespace) > 0 && tag.From.Namespace != stream.Namespace {
+		return "", false
+	}
+	ref := strings.TrimPrefix(tag.From.Name, stream.Name+":")
+	if strings.Contains(ref, ":") {
+		return "", false
+	}
+	return ref, true
+}
+
+var (
+	// ErrNoStreamRepository is returned if the status dockerImageRepository field was unset but the
+	// method required that value to create a pull spec.
+	ErrNoStreamRepository = fmt.Errorf("no image repository has been set on the image stream status")
+)
+
+// ResolvePullSpecForTag returns the correct pull spec for a given tag in an image stream. The correct
+// pull spec is the latest tagged, imported, or pushed image pull spec from the status, modified if
+// the Local referencePolicy is in place. This method assists clients in picking the right image pull
+// spec for a given scenario and handles the tag's referencePolicy replacement.
+//
+// If the spec tag is still in the process of importing the image or has failed to import the image,
+// the pull spec will be either the previous image in the tag history or the value of the spec tag.
+// Setting requireLatest to true will never return the previous image.
+//
+// If no spec tag is set and no history exists, the pull spec will be the internal image stream repository
+// value plus the tag value (<stream.status.dockerImageRepository>:<tag>). If defaultExternal is true
+// and stream.status.publicDockerImageRepository is not empty, that public repository spec will be
+// returned. If neither field is set ErrNoStreamRepository will be returned.
+//
+// hasStatus is true if the pull spec corresponds to an image in the status tags. hasImported is true
+// if an import is in progress for this tag. isLatest is true if the returned pull spec matches the
+// user's desired intent (will never be false when requireLatest is true).
+//
+// Use cases:
+//
+// * Get the latest pull spec representing user intent (user may be wrong or pull spec may not exist):
+//   - set requireLatest = true so we only consider the most recent pull/tag/import
+//   - if pullSpec is empty the tag doesn't represent an image
+//   - if isImporting is true the server may be having an issue getting this image or it doesn't exist
+//   - isLatest will always be true
+//
+// * Get the latest valid pull spec from the user (last good value):
+//   - set requireLatest = false so we take the last known valid pull/tag/import
+//   - if pullSpec is empty the tag doesn't represent a valid image
+//   - if isLatest is false, we chose an older image in the history
```

Comment:
Do we want to talk about `isImporting` for this case?  I expect that `isImporting` is more likely to be false than in the `requireLatest=true` case, but maybe we can still have `isImporting` false if we have never imported any images for this tag yet?

Reply:
isImporting is about the tag, not the value, so it has the same chance of being false in both scenarios.

Reply:
If there's an argument to make isImporting about the value that's potentially interesting I guess - since isLatest=false would still communicate importing anyway.  But that doesn't help us get rid of one of the returned booleans, so it seems like a wash.